ethportal-api/src/types/execution/header.rs

```diff
@@ -195,6 +197,73 @@ impl PartialEq for Header {
     }
 }
 
+impl From<Header> for RpcHeader {
+    fn from(header: Header) -> Self {
+        let hash = Some(header.hash().to_fixed_bytes().into());
+        let Header {
+            parent_hash,
+            uncles_hash,
+            author,
+            state_root,
+            transactions_root,
+            receipts_root,
+            logs_bloom,
+            difficulty,
+            number,
+            gas_limit,
+            gas_used,
+            timestamp,
+            extra_data,
+            mix_hash,
+            nonce: _,
+            base_fee_per_gas,
+            withdrawals_root,
+        } = header;
+
+        Self {
+            parent_hash: parent_hash.to_fixed_bytes().into(),
+            uncles_hash: uncles_hash.to_fixed_bytes().into(),
+            miner: author.to_fixed_bytes().into(),
+            state_root: state_root.to_fixed_bytes().into(),
+            transactions_root: transactions_root.to_fixed_bytes().into(),
+            receipts_root: receipts_root.to_fixed_bytes().into(),
+            logs_bloom: logs_bloom.to_fixed_bytes().into(),
+            difficulty: u256_to_uint256(difficulty),
+            number: Some(u64_to_uint256(number)),
+            gas_limit: u256_to_uint256(gas_limit),
+            gas_used: u256_to_uint256(gas_used),
+            timestamp: u64_to_uint256(timestamp),
+            extra_data: extra_data.into(),
+            mix_hash: mix_hash
+                .map(|mh| mh.to_fixed_bytes().into())
+                .unwrap_or_default(),
+            nonce: None,
+            base_fee_per_gas: base_fee_per_gas.map(u256_to_uint256),
+            withdrawals_root: withdrawals_root.map(|root| root.to_fixed_bytes().into()),
+            blob_gas_used: None,
+            excess_blob_gas: None,
+            hash,
+            parent_beacon_block_root: None,
+        }
+    }
+}
+
+fn u256_to_uint256(u256: U256) -> Uint<256, 4> {
```

Comment:
It seems like we could create our own custom wrapped type around `Uint<256, 4>` & then implement `From` rather than these conversion methods. Seems slightly cleaner, but also wrapped types can be annoying to manage, so I'm perfectly fine leaving this as is.

Reply:
Ok, I'll take a look. My intuition is that wrapping it just to unwrap it immediately is probably not worth the coding overhead, but I'll explore it a bit.

Reply:
```patch
diff --git a/ethportal-api/src/types/execution/header.rs b/ethportal-api/src/types/execution/header.rs
index e9407b8..91598ed 100644
--- a/ethportal-api/src/types/execution/header.rs
+++ b/ethportal-api/src/types/execution/header.rs
@@ -1,3 +1,5 @@
+use std::ops::Deref;
+
 use ethereum_types::{Bloom, H160, H256, H64, U256};
 use reth_rpc_types::Header as RpcHeader;
 use rlp::{Decodable, DecoderError, Encodable, Rlp, RlpStream};
@@ -232,7 +234,7 @@ impl From<Header> for RpcHeader {
             transactions_root: transactions_root.to_fixed_bytes().into(),
             receipts_root: receipts_root.to_fixed_bytes().into(),
             logs_bloom: logs_bloom.to_fixed_bytes().into(),
-            difficulty: u256_to_uint256(difficulty),
+            difficulty: *RethUint256::from(difficulty),
             number: Some(u64_to_uint256(number)),
             gas_limit: u256_to_uint256(gas_limit),
             gas_used: u256_to_uint256(gas_used),
@@ -252,6 +254,24 @@ impl From<Header> for RpcHeader {
     }
 }
 
+struct RethUint256(Uint<256, 4>);
+
+impl From<U256> for RethUint256 {
+    fn from(u256: U256) -> Self {
+        let mut bytes = [0u8; 32];
+        u256.to_big_endian(&mut bytes);
+        Self(Uint::from_be_bytes(bytes))
+    }
+}
+
+impl Deref for RethUint256 {
+    type Target = Uint<256, 4>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
 fn u256_to_uint256(u256: U256) -> Uint<256, 4> {
     let mut bytes = [0u8; 32];
     u256.to_big_endian(&mut bytes);
```

Not an obvious win to me. I think I'll leave it as-is unless you're seeing something I'm not about why it's better.